internal/service/dns_config/api_delegation_resource_test.go

```diff
@@ -0,0 +1,446 @@
+	var resourceName = "bloxone_dns_delegation.test_delegation_servers"
+	var v dns_config.ConfigDelegation
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:                 func() { acctest.PreCheck(t) },
+		ProtoV6ProviderFactories: acctest.ProtoV6ProviderFactories,
+		Steps: []resource.TestStep{
+			// Create and Read
+			{
+				Config: testAccDelegationDelegationServers("test.123.", "123.", "12.0.0.0", "ns1.com."),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckDelegationExists(context.Background(), resourceName, &v),
+					resource.TestCheckResourceAttr(resourceName, "delegation_servers.#", "1"),
+				),
+			},
+			// Update and Read
+			{
+				Config: testAccDelegationDelegationServers("test.123.", "123.", "12.0.0.1", "ns2.com."),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckDelegationExists(context.Background(), resourceName, &v),
+					resource.TestCheckResourceAttr(resourceName, "delegation_servers.#", "1"),
+				),
+			},
+			// Delete testing automatically occurs in TestCase
+		},
+	})
+}
+
+func TestAccDelegationResource_Disabled(t *testing.T) {
+	var resourceName = "bloxone_dns_delegation.test_disabled"
+	var v dns_config.ConfigDelegation
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:                 func() { acctest.PreCheck(t) },
+		ProtoV6ProviderFactories: acctest.ProtoV6ProviderFactories,
+		Steps: []resource.TestStep{
+			// Create and Read
+			{
+				Config: testAccDelegationDisabled("test.123.", "false", "123."),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckDelegationExists(context.Background(), resourceName, &v),
+					resource.TestCheckResourceAttr(resourceName, "disabled", "false"),
+				),
+			},
+			// Update and Read
+			{
+				Config: testAccDelegationDisabled("test.123.", "true", "123."),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckDelegationExists(context.Background(), resourceName, &v),
+					resource.TestCheckResourceAttr(resourceName, "disabled", "true"),
+				),
+			},
+			// Delete testing automatically occurs in TestCase
+		},
+	})
+}
+
+func TestAccDelegationResource_Fqdn(t *testing.T) {
+	var resourceName = "bloxone_dns_delegation.test_fqdn"
+	var v dns_config.ConfigDelegation
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:                 func() { acctest.PreCheck(t) },
+		ProtoV6ProviderFactories: acctest.ProtoV6ProviderFactories,
+		Steps: []resource.TestStep{
+			// Create and Read
+			{
+				Config: testAccDelegationFqdn("test.123.", "123."),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckDelegationExists(context.Background(), resourceName, &v),
+					resource.TestCheckResourceAttr(resourceName, "fqdn", "test.123."),
+				),
+			},
+			// Update and Read
+			{
+				Config: testAccDelegationFqdn("test1.123.", "123."),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckDelegationExists(context.Background(), resourceName, &v),
+					resource.TestCheckResourceAttr(resourceName, "fqdn", "test1.123."),
+				),
```

Comment:
can you check here if the previous object is destroyed here, since we use require_replace